### ISSUE TITLE
implement an interactive app for Raster Masking

### ISSUE THREAD

**Issue by janzandr** [EnMAP-Box/enmap-box#460]
_Discussed with @jakimowb and @thielfab._

Raster Masking includes 2 main steps: a) **Build Mask** and b) **Apply Mask**.

Different usecases for building a mask should be supported:
- thresholding, e.g. `array > 1000`, `array <= 1000`, `100 < array <= 1000`
- selecting categories from a classification, like implemented in GEE TSE app:
![image](https://user-images.githubusercontent.com/45030715/235084847-b7bcf515-206f-41d8-99b2-9eada2f9041e.png)
- selecting categories from a bitflag band, like implemented in GEE TSE app:
![image](https://user-images.githubusercontent.com/45030715/235085320-853270e7-cae3-4968-b7de-6eac53b47618.png)
- more usecases?

The mask building process should be interactive, so that after selecting a new threshold or a category, the updated mask is visualized inside a map view.

Optionally, instead of visualizing the mask, the user can select a raster to automatically apply the mask.

**Comment by aloboa** [comment]
Actually, Raster Data Plotting is very close to an ideal solution. Just
1. Add the option of saving the polygons that are drawn over the image whenever a polygon is drawn in the bivariate spectral plot.
2. For the univariate case (1 single band), display the histogram and let the user select segments on it.

**Comment by aloboa** [comment]
any news on implementing this tool?

**Comment by janzandr** [comment]
No, but some aspects will be implemented for specific sensors here #1032.
In later versions of #1032, you will be able to do pixel quality screening on-the-fly (probably not in the first version).
